schematools/types.py

```diff
@@ -73,19 +74,74 @@ def from_dict(cls, obj: dict):
     @property
     def tables(self) -> typing.List[DatasetTableSchema]:
         """Access the tables within the file"""
-        return [DatasetTableSchema(i, _parent_schema=self) for i in self["tables"]]
+        tables = []
+        for table_schema in self["tables"]:
+            table = DatasetTableSchema(table_schema, _parent_schema=self)
+            tables.append(table)
+        return tables
+
+    def get_tables(self, include_nested=False) -> typing.List[DatasetTableSchema]:
+        """List tables, including nested"""
+        tables = self.tables
+        if include_nested:
+            tables += self.nested_tables
+        return tables
 
     def get_table_by_id(self, table_id: str) -> DatasetTableSchema:
-        for table in self["tables"]:
-            if table["id"] == table_id:
-                return DatasetTableSchema(table, _parent_schema=self)
+        for table in self.get_tables(include_nested=True):
+            if table.id == table_id:
+                return table
```

Comment:
aub niet terugdraaien. Dit had ik juist geschreven zodat we niet eerst alle tables omzetten naar `DatasetTableSchema` en vervolgens weer weggooien.

Reply:
Maar dat is niet meer mogelijk, omdat Nested Tabellen zijn gebouwd op basis van de `DatasetTableSchema`s.

Dat kan zeker veranderd worden, maar is dat echt noodzakelijk? de overhead is nog niet zo groot.